src/routes/docs/tutorials/nuxt/step-4/+page.markdoc

```diff
@@ -0,0 +1,175 @@
+
+We will build a page with a simple login form and store its related logic in a `useUserSession` composable so it can be reused, starting with the composable.
+
+# User session composable {% #user-session-composable %}
+
+There are a few standard functions involved in handling a user session that is added to the composable.
+The user needs to be able to register to an account, login to the account and logout from it.
+
+We are using Appwrite as a backend to handle the user details, so we need to connect to Appwrite by importing the configurations from step 3.
+The response from these interactions will be stored as references to get more information about the user in our app.
+
+In your `composable` directory, create the file `useUserSession.js` and add the following code.
+Then you can call the  `useUserSession()` function in the pages and components to use the functionality.
+
+```js
+// composable/useUserSession.js
+
+import { ID } from "appwrite";
+import { ref } from "vue";
+import { account } from "/appwrite";
+
+const current = ref(null) //Reference to current user object
+const isLoggedIn = ref(null) //Reference to check user status
+
+export const useUserSession = () => {
+  const register = async (email, password) => {
+    await account.create(ID.unique(), email, password); // Register new user in Appwrite
+    await login(email, password); // Login registered user
+  };
+
+  const login = async (email, password) => {
+    const authUser = await account.createEmailSession(email, password); // Open user session in Appwrite
+    current.value = authUser; // Pass user data to current ref
+    isLoggedIn.value = true; // Set ref to true
+    navigateTo("/");
+  };
+
+  const logout = async () => {
+    await account.deleteSession("current"); // Delete Appwrite user session
+    current.value = null; // Clear current ref
+    isLoggedIn.value = false; // Set ref to false
+    navigateTo("/");
+  };
+
+  return {
+    current,
+    isLoggedIn,
+    login,
+    logout,
+    register,
+  };
+};
+
+```
+# Login page {% #login-page %}
+
+To get the user input we will build a simple form with the logic from the `useUserSession()` composable.
+The form will have two input fields, one for the user's email-address and one for the password.
+Underneath will be two buttons for logging in or registering the user.
+
+Create a new file in the `pages` directory called `login.vue`.
+This will not only create a new page, it will also add the route `/login` to the url.
+In step 5 we will add a loginbutton that will redirect us to this page.
+Add the following code to build the form.
+
+```vue
+// pages/login.vue
+
+<template>
+  <div class="u-max-width-650" style="margin: 0 auto;">
+    <section class="card u-margin-32">
+      <h2 class="eyebrow-heading-2">Login/Register</h2>
+      <form
+        class="form u-width-full-line u-max-width-500 u-margin-block-start-16"
+        @submit.prevent="handleLogin || handleRegistration"
```

Comment:
This is problematic. I assume this tries to register in case a login failed. What if email was input wrongly? IT will create a new user just because of that, which is quite odd. Should be separate buttons/forms.

Reply:
This is actually just the `event.preventDefault` for these handlers. This does not call handle registration if you have wrong email. It stops event propagation.

Reply:
I wans't talking about the .prevent, but the `||`.

But it's worse, this doesn't seem to do anything at all. i tried reproducing in StackBlitz: https://stackblitz.com/edit/github-wlmexm?file=app.vue

Reply:
Re-reading the code example, each button has its own onClick handler. So the submit is only there for prevent default. This is quite non-standard too, since this now won't work when pressing `Enter`.

Alternatives:
- Two separate forms (most standard)
- A switch/checkbox that indicates whether you're logging in or registering (weird tho)
- Some type of smart login flow where if login fails, it asks if you want to register.